src/core/public/chrome/nav_group/nav_group_service.ts

```diff
@@ -115,12 +123,87 @@ export class ChromeNavGroupService {
   }
 
   private sortNavGroupNavLinks(
-    navGroup: NavGroupItemInMap,
+    navLinks: NavGroupItemInMap['navLinks'],
     allValidNavLinks: Array<Readonly<ChromeNavLink>>
   ) {
-    return getSortedNavLinks(
-      fulfillRegistrationLinksToChromeNavLinks(navGroup.navLinks, allValidNavLinks)
-    );
+    return getSortedNavLinks(fulfillRegistrationLinksToChromeNavLinks(navLinks, allValidNavLinks));
+  }
+
+  private getNavLinksForAllUseCase(
+    navGroupsMap: Record<string, NavGroupItemInMap>,
+    navLinks: Array<Readonly<ChromeNavLink>>
+  ) {
+    const navLinksResult: ChromeRegistrationNavLink[] = [];
+
+    // Append all the links that do not have use case info to keep backward compatible
+    const linkIdsWithNavGroupInfo = Object.values(navGroupsMap).reduce((total, navGroup) => {
+      return [...total, ...navGroup.navLinks.map((navLink) => navLink.id)];
+    }, [] as string[]);
+    navLinks.forEach((navLink) => {
+      if (linkIdsWithNavGroupInfo.includes(navLink.id)) {
+        return;
+      }
+      navLinksResult.push({
+        ...navLink,
+        category: customCategory,
+      });
+    });
+
+    // Append all the links registered to all use case
+    navGroupsMap[ALL_USE_CASE_ID]?.navLinks.forEach((navLink) => {
+      navLinksResult.push(navLink);
+    });
+
+    // Append use case section into left navigation
+    Object.values(navGroupsMap).forEach((group) => {
+      if (group.type) {
+        return;
+      }
+      const categoryInfo = {
+        id: group.id,
+        label: group.title,
+        order: group.order,
+      };
+
+      const fulfilledLinksOfNavGroup = fulfillRegistrationLinksToChromeNavLinks(
+        group.navLinks,
+        navLinks
+      );
+
+      const linksForAllUseCaseWithinNavGroup: ChromeRegistrationNavLink[] = [];
+
+      fulfilledLinksOfNavGroup.forEach((navLink) => {
+        if (!navLink.showInAllNavGroup) {
+          return;
+        }
+
+        linksForAllUseCaseWithinNavGroup.push({
+          ...navLink,
+          category: categoryInfo,
+        });
+      });
+
+      navLinksResult.push(...linksForAllUseCaseWithinNavGroup);
```

Comment:
for those links that will exist in multiple use cases like visualization, will navLinksResult have duplicate records?

Reply:
It can be but those nav links should have different categories and it is supposed that only the special features within a use case should be declared as `showInAllNavGroup`. Let's bear with it.